cdap-app-fabric/src/main/java/io/cdap/cdap/app/store/Store.java

```diff
@@ -368,6 +368,19 @@ List<ProgramHistory> getRuns(Collection<ProgramReference> programs, ProgramRunSt
    */
   int addApplication(ApplicationId id, ApplicationMeta meta) throws ConflictException;
 
+  /**
+   * Creates new application if it doesn't exist. Updates existing one otherwise.
+   * Marks the application as latest based on the isLatest param.
+   *
+   * @param id            application id
+   * @param meta          application metadata to store
+   * @param isLatest      boolean, indicating if the application should be marked latest
+   * @return              the number of edits to the application. A new application will return 0.
+   * @throws ConflictException if the app cannot be deployed when the user provided parent-version doesn't match the
+   *     current latest version
+   */
+  int addApplication(ApplicationId id, ApplicationMeta meta, boolean isLatest) throws ConflictException;
```

Comment:
How is this one different than the one above that doesn't take a `isLatest`? Ideally shouldn't use boolean. parameter to control behavior, but rather have a different method.

Reply:
@chtyim 
The method `addApplication(ApplicationId id, ApplicationMeta meta)` adds an application and always marks it as latest.

Whereas the method `addApplication(ApplicationId id, ApplicationMeta meta, boolean isLatest)` marks the added application as latest based on the value of `isLatest`.

I agree that, having a different method name would be better. In the latest commit I have renamed (formerly) `addApplication(ApplicationId id, ApplicationMeta meta)` to `addLatestApplication` for clarity.